pkg/cmd/server/start/start_master.go

```diff
@@ -594,31 +594,41 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 		controllerManagerOptions = kc.ControllerManager
 	}
 
-	// Start these first, because they provide credentials for other controllers' clients
-	oc.RunServiceAccountsController()
-	oc.RunServiceAccountTokensController(controllerManagerOptions)
-	// used by admission controllers
-	oc.RunServiceAccountPullSecretsControllers()
-	oc.RunSecurityAllocationController()
-
-	_, _, _, binderClient, err := oc.GetServiceAccountClients(bootstrappolicy.InfraPersistentVolumeBinderControllerServiceAccountName)
-	if err != nil {
-		glog.Fatalf("Could not get client for persistent volume binder controller: %v", err)
+	rootClientBuilder := controller.SimpleControllerClientBuilder{
+		ClientConfig: &oc.PrivilegedLoopbackClientConfig,
 	}
 
-	_, _, _, attachDetachControllerClient, err := oc.GetServiceAccountClients(bootstrappolicy.InfraPersistentVolumeAttachDetachControllerServiceAccountName)
-	if err != nil {
-		glog.Fatalf("Could not get client for attach detach controller: %v", err)
+	// openshiftControllerPreContext represents a context used to start controllers that
+	// requires privileged 'rootClientBuilder' iow. have to start before any other
+	// controller starts (including kubernetes controllers).
+	openshiftControllerContext := origincontrollers.ControllerContext{
+		KubeControllerContext: kctrlmgr.ControllerContext{Options: *controllerManagerOptions},
+		ClientBuilder: origincontrollers.OpenshiftControllerClientBuilder{
+			ControllerClientBuilder: controller.SAControllerClientBuilder{
+				ClientConfig:         restclient.AnonymousClientConfig(&oc.PrivilegedLoopbackClientConfig),
+				CoreClient:           oc.PrivilegedLoopbackKubernetesClientsetExternal.Core(),
+				AuthenticationClient: oc.PrivilegedLoopbackKubernetesClientsetExternal.Authentication(),
+				Namespace:            bootstrappolicy.DefaultOpenShiftInfraNamespace,
+			},
+		},
+		TemplateInformers:            oc.TemplateInformers,
```

Comment:
I think we'll want a struct for our informers eventually, but this moves us forward.